### ISSUE TITLE
guarddog fails to parse the whole requirements.txt after encountering package name starting with git+https:// 

### ISSUE THREAD

**Issue by maciejstromich** [DataDog/guarddog#88]
requirements.txt
```
black==22.3.0
git+https://github.com/PyCQA/bandit.git@main
cfn-lint
hypothesis
```

docker-compose.yml
```
  guarddog:
    image: ghcr.io/datadog/guarddog:v0.1.4
    volumes:
      - ./:/code
```
execution:
```
docker-compose run  guarddog verify /code/requirements.txt
Received error Parse error at "'+https:/'": Expected string_endFound 0 potentially malicious indicators scanning black version 22.3.0
```

as seen in the output anything below git+https:// is not being checked for malicious indicators. 
also upon parse error the output is broken and doesn't contain break line between string_end and Found

**Comment by vdeturckheim** [comment]
Thanks for the report @maciejstromich . I have been digging on this one and I think the fix is non-trivial.
This syntax is actually accepted by `pip` but not by `pkg_resources` that we use to parse the requirement files.
 
I am currently considering the solutions for this one and it could take a couple days.

**Comment by zmallen** [comment]
Hey there!

According to https://pip.pypa.io/en/stable/reference/requirement-specifiers/, url-based requirements specifiers require the package name/egg name in front of a url

I replicated your error locally, and then got a successful run by replacing with the following:

```text
black==22.3.0
bandit @ git+https://github.com/PyCQA/bandit.git@main
cfn-lint
hypothesis
```

```bash
└> python -m guarddog verify example-requirements.txt

Found 0 potentially malicious indicators scanning black version 22.3.0

Found 0 potentially malicious indicators scanning bandit

Found 0 potentially malicious indicators scanning cfn-lint

Found 0 potentially malicious indicators scanning hypothesis
```

Can you try that?

**Comment by maciejstromich** [comment]
yeah that works! Thanks for your input in making my codebases better 👍

**Comment by zmallen** [comment]
Yeah you are definitely right. I think this is an issue with `pkg_resources`. Pip has some magic to get around this, but they specifically don't have an API that lets you do this

**Comment by vdeturckheim** [comment]
Exactly, I think for now, we will log a specific warning for unsupported syntaxes

**Comment by christophetd** [comment]
@maciejstromich Can you let us know what you think of the behavior introduced in #96?

**Comment by maciejstromich** [comment]
I guess it's fine as long as the whole requirements file gets processed. I can update my files and put package_name @ before so they get checked correctly.

**Comment by christophetd** [comment]
Released as part of v0.1.7 https://github.com/DataDog/guarddog/releases/tag/v0.1.7
